hax/treemakers/CorrectedDoubleS1Scatter.py

```diff
@@ -0,0 +1,336 @@
+    - INT_y_pax: The y-position of this interaction
+    - INT_z_pax: the z-position of this interaction 
+    
+    - INT_z_observed: The z-position of this interaction without correction
+    - INT_drift_time : The drift time of the interaction
+    
+    - INT_x_nn : The x-position of this interaction with NeutralNetwork Analysis
+    - INT_y_nn : The y-position of this interaction with NeutralNetwork Analysis 
+    - INT_r_nn : The r-position of this interaction with NeutralNetwork Analysis  
+
+    ### Corrected Signal Output (for INT in [int_a, int_b]):
+    # Data-driven 3D position correction 
+    - INT_r_correction_3d_nn : correction value for r position using NN
+    - INT_r_3d_nn : The corrected interaction r coordinate using NN 
+    - INT_x_3d_nn : The corrected interaction x coordinate using NN 
+    - INT_y_3d_nn : The corrected interaction y coordinate using NN 
+    - INT_z_correction_3d_nn : correction value for z position using NN
+    - INT_z_3d_nn : The corrected interaction z coordinate using NN 
+    
+    # LCE correction on S1 using NN FDC xyz-corrected position
+    - s1_INT_xyz_correction_nn_fdc_3d : correction value for s1 signals using the INT_nn position :
+    
+    /!\ Two way of doing things for the S1_b signal : 
+      - either used the int_a position to correct s1_b signal (since S1_a and S1_b are closed in time) : by default
+      - either used the int_b position to correct s1_b signal (but most of the time the s2_b signal (and thus z position) is badly reconstructed )
+    
+    - cS1_a and cS1_b : the corrected s1_a signal using int_a_3d_nn corrected position
+    - cS1_b_int_b: the corrected s1_b signal using int_b_3d_nn corrected position
+    
+    ### DoubleScatter Specific Output:
+    - ds_s1_b_n_distinct_channels: number of PMTs contributing to s1_b distinct from the PMTs that contributed to s1_a
+    - ds_s1_dt : delay time between s1_a_center_time and s1_b_center_time
+    - ds_second_s2: 1 if selected interactions have distinct s2s
+    """
+    
+    __version__ = '0.2'
+    extra_branches = ['peaks.n_contributing_channels',
+                      'peaks.center_time',
+                      'peaks.s2_saturation_correction',
+                      'interactions.s2_lifetime_correction',
+                      'peaks.area_fraction_top',
+                      'peaks.area',
+                      'peaks.reconstructed_positions*',
+                      'interactions.x',
+                      'interactions.y',
+                      'interactions.z',
+                      'interactions.r_correction',
+                      'interactions.z_correction',
+                      'interactions.drift_time',
+                      'start_time',
+                      'peaks.hits*',
+                      'interactions.s1_pattern_fit'
+                     ]
+    extra_metadata = hax.config['corrections_definitions']
+    
+    # Load Correction Map
+    loaded_3d_fdc_map_name = None
```

Comment:
Should we make a common "maps" class that takes care of all of these at hax init? (Reducing the duplicated functions below also in ```corrections.py```.)

Reply:
I tried to but I am still quite new in Python and I didn't find a way to make it works

Reply:
@pdeperio yes, but considering the time sensitivity we thought to do it after. For example next time we need to make an update we can just request a refactor, hopefully at a point when we can afford to spend the extra week